src/abstraction/transient.rs

```diff
@@ -456,3 +476,30 @@ impl Default for TransientKeyContextBuilder {
         TransientKeyContextBuilder::new()
     }
 }
+
+/// Parameters for the kinds of keys supported by the context
+#[derive(Debug, Clone, Copy)]
+pub enum KeyParams {
+    Rsa {
+        /// Size of key in bits
+        ///
+        /// Can only be one of: 1024, 2048, 3072 or 4096
+        size: u16,
```

Comment:
Could we have a `RsaSize` enumeration for those four possible values so that the check line 104 is no longer needed?

Reply:
Yeah, I know, tried that with `Cipher` some time ago, but it feels overkill tbh... Could change if you want. Also the variant names feel weird af because you can't start a name with digits.

Reply:
I think it is worth to do it!
We could write the number in clear letters like:
```rust
enum RsaSize {
    OneThousandTwentyFour,
    TwoThousandFortyEight,
    ...
```

(joking)

Actually it could be possible

Reply:
Yeah, for the symmetric ones I went for `Bits128`, `Bits192`... but anyway... I'll skip it for now